### DISCUSSION TITLE
Option to start minimised or to minimise to system tray (appimage/flatpak)

### DISCUSSION THREAD

**Discussion by Danie10** [rustdesk/rustdesk#8098] [General]
I have an issue with remote support for my family as invariably when Rustdesk autostarts, it gets closed with the X on the window, and this closes the app altogether. Then, when I need to later remote access to shut the PC down, do updates late at night, etc, there is no Rustdesk service running.

Is it possible to either:
1. Have a settings tick option or runtime parameter to start the Flatpak service minimised?
2. Or to have a settings option that minimises the window instead of closing it when the X is clicked on the window?

Either option would work fine. It can still show in the task bar etc so need not be hidden at all.

**Comment by rustdesk** [comment]
We have no plan, but PR is welcome

**Reply by DarkLordGMS** [reply]
So if we're using Linux Mint we can't use RustDesk without opening it first?

**Reply by xlionjuan** [reply]
You should not install sandboxed version of a remote desktop software and hope it has limited power.

**Reply by kvnis** [reply]
> So if we're using Linux Mint we can't use RustDesk without opening it first?

Hello from the future! I am fairly new to Linux (just moved from Windows to PopOS, which basically is Ubuntu).
I had the same issue, where I had to open RustDesk application every time I wanted to connect to it.

The issue for me was that I installed RustDesk using "PopOS! Store", which is the PopOS equilavant of Ubuntu Software Store (where you can install applications, much like App Store or Play Store on your phone). Apparently the version I downloaded was a FlatPak, which does not use systemd (systemctl).

The solution were to simply uninstall the current version I had installed, then download and install RustDesk from their website using the .deb file instead. This creates a systemd for that, making it run as a service (you do not have to open the software every time to connect to the PC).

Hope this helps!

**Reply by xlionjuan** [reply]
The appstore version would be flatpak

**Reply by monx-spec** [reply]
> We have no plan, but PR is welcome

As for no, still no plan? Launching Rustdesk minimized would help numerous non-tech-savvy users.

**Comment by Badjowk** [comment]
Hey everyone! Since the rustdesk team has no plan/time to implement this feature, I finally found a workaround if you are using KDE Plasma with the rustdesk Flatpak. Its not perfect but it beats having the rustdesk window show as constantly open and active. 

Step 1: Setup rustdesk to Autostart using the KDE settings menu
<img width="360" height="54" alt="Screenshot_20260115_000110" src="https://github.com/user-attachments/assets/eab7392b-4d2b-49b7-a4e3-1d3d0bb8bae5" />

Step 2: Still in the settings app, go to Window Management -> Window Rules

Step 3: Make a new rule with the following settings:
<img width="654" height="558" alt="Screenshot_20260115_000724" src="https://github.com/user-attachments/assets/8ceac2c8-3bf7-4871-8425-11ca0877255e" />



Step 4: Success. rustdesk will now start at login, and as long as you minimize(not close) rustdesk, it will continue to run in the background and not be in the way or feel like clutter. hope this helps someone.

**Reply by xdpirate** [reply]
After some fiddling, I've found an alternative solution that gives you a tray icon and lets you use the program as normal, both for incoming and outgoing connections. This requires you to be running an X11 session, as KDocker doesn't work on Wayland. [kwin-minimize2tray](https://github.com/luisbocanegra/kwin-minimize2tray) may work on Wayland, but I have not tested it.

1. Install KDocker (`sudo apt install kdocker` or get from [its repository](https://github.com/user-none/KDocker))
2. Create a script, e.g. `/opt/scripts/dock-rustdesk.sh`:

```
#!/usr/bin/env bash
flatpak run com.rustdesk.RustDesk &
kdocker -n RustDesk &
disown
```

3. Add that script to your autostart UI (or to a `.desktop` file in `~/.config/autostart`)
4. When you log in, RustDesk will flash for a moment then get placed in the tray thanks to KDocker

Example desktop file for autostart (`~/.config/autostart/RustDesk.desktop`):

```
[Desktop Entry]
Exec=/opt/scripts/dock-rustdesk.sh
Name=Run and dock RustDesk
Type=Application
X-KDE-AutostartScript=true
```

Remember to `chmod +x /opt/scripts/dock-rustdesk.sh ~/.config/autostart/RustDesk.desktop`.

After docking, you may notice that the tray icon is a generic one, you can fix that by right-clicking it, going to Options > Set icon, then picking an icon for it. I downloaded the [RustDesk icon from this repo](https://github.com/rustdesk/rustdesk/blob/master/res/128x128.png) (click the "Download raw file" button on the right) and saved it to `/opt/icons` (the file isn't copied to a local store so it needs to exist for the icon to persist through relaunches).

Finally, right click the icon, go to Options > Save settings > Rustdesk only, to persist your icon (and any other settings you change).

Click the tray icon to toggle the RustDesk window. When you minimise RustDesk, it will be hidden behind the tray icon. If you close the RustDesk window, it will exit.
